samples/azure-quantum/qsp/qsp.ipynb

```diff
@@ -0,0 +1,615 @@
+          },
+          "metadata": {}
+        }
+      ],
+      "execution_count": 7,
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## 2. Simulating the QSP polynomial with a quantum circuit\n",
+        "\n",
+        "Next, we are creating the quantum circuit using $W$ and $S$ gates that is supposed to implement our target unitary with the QSP polynomial $P$ in the top-left corner."
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def qsp_circuit(x, phases):\n",
+        "    \"\"\"Creates a single-qubit unitary for some given QSP phases\"\"\"\n",
+        "\n",
+        "    # Map SumPy phases into NumPy phases and reverse them (gate application\n",
+        "    # direction is opposite to matrix multiplication)\n",
+        "    np_phases = [float(phi) for phi in phases]\n",
+        "    np_phases.reverse()\n",
+        "\n",
+        "    circ = QuantumCircuit(1)\n",
+        "\n",
+        "    # We need to apply gates in reverse order to matrix multiplication\n",
+        "    circ.append(SGate(np_phases[0]), [0])  # apply last angle\n",
+        "    for phi in np_phases[1:]:           # iterate through all but last angle in reverse order\n",
+        "        circ.append(WGate(x), [0])\n",
+        "        circ.append(SGate(phi), [0])\n",
+        "\n",
+        "    return circ"
+      ],
+      "outputs": [],
+      "execution_count": 8,
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "We simulate the unitary matrices of QSP circuits constructed for some random values of $x \\in [-1, 1]$ and compare them to the unitary $U$ that we explicitly constructed above."
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Test QSP polynomial\n",
+        "for _ in range(10):\n",
+        "    x = random.rand() * 2.0 - 1.0\n",
+        "\n",
+        "    # Create circuit\n",
+        "    circ = qsp_circuit(x, phases)\n",
+        "\n",
+        "    # Run circuit and return unitary matrix\n",
+        "    actual = execute(circ, unitary_backend).result().get_unitary()\n",
+        "\n",
+        "    # Expected result from U by plugging in x value for x symbol (transforms SymPy matrix into NumPy array)\n",
+        "    expected = np.array(U.evalf(subs={'x': x})).astype(complex)\n",
+        "\n",
+        "    assert np.allclose(actual, expected)"
+      ],
+      "outputs": [],
+      "execution_count": 9,
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Great, the test passes! This means we should be able to reconstruct the polynomial $|P(x)|^2$ that we plotted above by running the QSP circuit several times and estimate the measurement probability of receiving $|0\\rangle$ from various measurement results iterating through different values for $x$.\n",
```

Comment:
Is estimating the measurement probability part of the way to reconstruct the polynomial, or something we're able to do in addition to reconstructing it? If it's the former, it should be "estimating"

Reply:
```suggestion
        "Great, the test passes! This means we should be able to reconstruct the polynomial $|P(x)|^2$ that we plotted above by running the QSP circuit several times and estimating the measurement probability of receiving $|0\\rangle$ from various measurement results iterating through different values for $x$.\n",
```